atom/browser/importer/profile_writer.cc

```diff
@@ -1,114 +1,49 @@
-// #include "chrome/browser/password_manager/password_store_factory.h"
 #include "chrome/browser/profiles/profile.h"
-// #include "chrome/browser/search_engines/template_url_service_factory.h"
-// #include "chrome/browser/web_data_service_factory.h"
 #include "chrome/common/importer/imported_bookmark_entry.h"
-#include "chrome/common/pref_names.h"
 #include "components/autofill/core/browser/webdata/autofill_entry.h"
 #include "components/autofill/core/common/password_form.h"
-// #include
-// "components/autofill/core/browser/webdata/autofill_webdata_service.h"
-#include "components/bookmarks/browser/bookmark_model.h"
-// #include "components/bookmarks/common/bookmark_pref_names.h"
-// #include "components/favicon/core/favicon_service.h"
-#include "components/history/core/browser/history_service.h"
-// #include "components/password_manager/core/browser/password_store.h"
-#include "components/prefs/pref_service.h"
-#include "components/search_engines/template_url.h"
-#include "components/search_engines/template_url_service.h"
 
 #if defined(OS_WIN)
-// #include "chrome/browser/web_data_service_factory.h"
-// #include "components/password_manager/core/browser/webdata/\
-// password_web_data_service_win.h"
+#include "components/password_manager/core/browser/webdata/password_web_data_service_win.h"
 #endif
-/*
-using bookmarks::BookmarkModel;
-using bookmarks::BookmarkNode;
-
-namespace {
-
-// Generates a unique folder name. If |folder_name| is not unique, then this
-// repeatedly tests for '|folder_name| + (i)' until a unique name is found.
-base::string16 GenerateUniqueFolderName(BookmarkModel* model,
-                                        const base::string16& folder_name) {
-  // Build a set containing the bookmark bar folder names.
-  std::set<base::string16> existing_folder_names;
-  const BookmarkNode* bookmark_bar = model->bookmark_bar_node();
-  for (int i = 0; i < bookmark_bar->child_count(); ++i) {
-    const BookmarkNode* node = bookmark_bar->GetChild(i);
-    if (node->is_folder())
-      existing_folder_names.insert(node->GetTitle());
-  }
-
-  // If the given name is unique, use it.
-  if (existing_folder_names.find(folder_name) == existing_folder_names.end())
-    return folder_name;
-
-  // Otherwise iterate until we find a unique name.
-  for (size_t i = 1; i <= existing_folder_names.size(); ++i) {
-    base::string16 name = folder_name + base::ASCIIToUTF16(" (") +
-        base::SizeTToString16(i) + base::ASCIIToUTF16(")");
-    if (existing_folder_names.find(name) == existing_folder_names.end())
-      return name;
-  }
 
-  NOTREACHED();
-  return folder_name;
-}
-
-// Shows the bookmarks toolbar.
-void ShowBookmarkBar(Profile* profile) {
-  //profile->GetPrefs()->SetBoolean(bookmarks::prefs::kShowBookmarkBar, true);
-}
+namespace importer {
+void ShowImportLockDialog(gfx::NativeWindow parent,
+                          const base::Callback<void(bool)>& callback) {}
```

Comment:
@bridiver I think this should call `ProfileWriter::ShowWarningDialog` to emit event.